pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -456,3 +459,19 @@ func generateOwnershipUniqueKeys(owner hivev1.MetaRuntimeObject) []*controllerut
 		},
 	}
 }
+
+// deleteInstallJob deletes the install job of a successful provision
+func (r *ReconcileClusterProvision) deleteInstallJob(provision *hivev1.ClusterProvision, pLog log.FieldLogger) (reconcile.Result, error) {
+	pLog.Info("deleting successful install job")
+	job := &batchv1.Job{}
+	if err := r.Get(context.TODO(), types.NamespacedName{Name: provision.Status.JobRef.Name, Namespace: provision.Namespace}, job); err != nil {
+		pLog.WithError(err).Log(controllerutils.LogLevel(err), "could not get install job")
+		return reconcile.Result{}, err
```

Comment:
Do not return an error when the job has already been deleted.

Reply:
I think returning error here is fine since we will executing this code only once. Correct me If I am wrong

Reply:
If we return an error here when the job has already been deleted, then `provision.Status.JobRef` will never be cleared.